### PR TITLE
Closes #2608: Remove SpringRestDocs as it has been replaced with OpenAPI

### LINKED ISSUE #2608: Remove SpringRestDocs as it has been replaced with OpenAPI
Things to remove:
- SpringRestDocTests (and make sure, there is a corresponding integration test to keep up the test coverage)
- Maven Dependencies
- Java Docs? E.g. on the controller level
- more ?

### PR DESCRIPTION
**Pull request by jamesrdi**
### Thanks for your PR! Please fill out the following list :)

---
- [ ] I put the ticket or multiple tickets in review
- [ ] Commit message format → Closes #&lt;Issue Number&gt; - Your commit message.
- [ ] Sonarcloud link : \<add the link here>
- [x] No documentation update needed
- [ ] Link to PR with documentation update: \<add the link here>
- [x] No Release Notes needed
- [ ] Release Notes :

```

```

### CODE REVIEW

**Comment by jamesrdi** [comment]
Should wait until #2622 is merged

File: rest/taskana-rest-spring/src/main/java/pro/taskana/workbasket/rest/models/WorkbasketSummaryRepresentationModel.java [hunk 1]
```diff
@@ -92,19 +66,19 @@ public class WorkbasketSummaryRepresentationModel
...
 
-  /** The third Org Level. */
   @Schema(name = "orgLevel3", description = "The third Org Level.")
   protected String orgLevel3;
 
-  /** The fourth Org Level (the lowest one). */
   @Schema(name = "orgLevel4", description = "The fourth Org Level.")
   protected String orgLevel4;
 
   /** Identifier to tell if this workbasket can be deleted. */
```
Comment (holgerhagen) [inline rest/taskana-rest-spring/src/main/java/pro/taskana/workbasket/rest/models/WorkbasketSummaryRepresentationModel.java:78]: this line needs to be deleted as well?

**Review by holgerhagen — CHANGES_REQUESTED** [review]

File: rest/taskana-rest-spring/src/main/java/pro/taskana/workbasket/rest/models/WorkbasketSummaryRepresentationModel.java (hunk 1 above)
Comment (jamesrdi) [reply rest/taskana-rest-spring/src/main/java/pro/taskana/workbasket/rest/models/WorkbasketSummaryRepresentationModel.java:78]: Yes you are right, I will delete it

File: history/taskana-simplehistory-rest-spring/src/main/java/pro/taskana/simplehistory/rest/TaskHistoryQueryFilterParameter.java [hunk 2]
```diff
@@ -12,234 +12,109 @@
 
 public class TaskHistoryQueryFilterParameter implements QueryParameter<TaskHistoryQuery, Void> {
 
-  /** Filter by the event type of the Task History Event. This is an exact match. */
   @JsonProperty("event-type")
   private final String[] eventType;
 
-  /**
-   * Filter by the event type of the Task History Event. This results in a substring search.. (% is
```
[the remark this answers is not in the document]
Comment (jamesrdi) [reply history/taskana-simplehistory-rest-spring/src/main/java/pro/taskana/simplehistory/rest/TaskHistoryQueryFilterParameter.java:20]: This information is copied as OpenAPI annotation in #2622, that is why I have to wait until #2622 is merged, then I will rebase it

**Pull request closed without merging**